render/src/lib/views/sidebar.vertical/port.available/components.ts

```diff
@@ -0,0 +1,304 @@
+                this._read = message.load[this.port.path];
+            } else if (message.event === EHostEvents.state && message.state[this.port.path]) {
+                this._read = message.state[this.port.path].ioState.read;
+            }
+        });
+    }
+
+    ngOnInit() {
+        this._session = Service.getSessionID();
+        this._defaultOptions.path = this.port.path;
+    }
+
+    ngOnDestroy() {
+        if (this._chart) {
+            this._chart.destroy();
+            this._chart = undefined;
+        }
+
+        Object.keys(this._subscriptions).forEach((key: string) => {
+            this._subscriptions[key].unsubscribe();
+        });
+        this._destroyed = true;
+    }
+
+    private _forceUpdate() {
+        if (this._destroyed) {
+            return;
+        }
+        this._cdRef.detectChanges();
+    }
+
+    private _loadSession() {
+        const cSessionPort = this._getSessionPort();
+        this._chart.config.options.animation.duration = 5000;
+        if (cSessionPort) {
+            this._ng_isConnected = cSessionPort.connected;
+            this._chart.config.data.datasets[0].data = cSessionPort.sparkline_data.slice(0, Service.chart_limit + 1);
+            this._chart.config.data.labels = this._chart_labels.slice(0, Service.chart_limit);
+
+            if (cSessionPort.spying === false) {
+                Service.startSpy([this._defaultOptions]).then(() => {
+                    cSessionPort.spying = true;
+                }).catch((error: Error) => {
+                    Service.notify('Error', error.message, ENotificationType.error);
+                });
+            }
+        } else {
+            console.error('Something went wrong while loading the session');
+        }
+    }
+
+    private _formatLoad(load: number): string {
+        let read: string = '';
+        if (load > 1024 * 1024 * 1024) {
+            read = (load / 1024 / 1024 / 1024).toFixed(2) + ' Gb';
+        } else if (load > 1024 * 1024) {
+            read = (load / 1024 / 1024).toFixed(2) + ' Mb';
+        } else if (load > 1024) {
+            read = (load / 1024).toFixed(2) + ' Kb';
+        } else {
+            read = load + ' b';
+        }
+        return read;
+    }
+
+    private _createChart() {
+        this._canvas = this.canvases.find(canvas => canvas.nativeElement.id === `canvas_${this.port.path}`);
+        this._ctx = this._canvas.nativeElement.getContext('2d');
+
+        const cSessionPort = this._getSessionPort();
+        if (cSessionPort) {
+            const cSparklineData = cSessionPort.sparkline_data;
+            this._chart = new Chart(this._ctx, {
+                type: 'line',
+                data: {
+                    labels: this._chart_labels.slice(0, Service.chart_limit),
+                    datasets: [{
+                        data: cSparklineData.slice(cSparklineData.length - Service.chart_limit),
+                        borderColor: Service.portColor[this.port.path],
```

Comment:
In general it isn't good idea to share props like here `Service.portColor`. Better keep it private and use something like `Service.getPortColor(path:string): string | undefined`